thing-url-adapter.js

```diff
@@ -798,10 +839,34 @@ function loadThingURLAdapter(addonManager) {
     if (typeof config.pollInterval === 'number') {
       adapter.pollInterval = config.pollInterval * 1000;
     }
-
     for (const url of config.urls) {
-      adapter.loadThing(url);
+      if ('authentication' in url) {
+        // remove http(s) from url
+        let url_stub = '';
+        if (url.href.includes('http://')) {
+          url_stub = url.href.substr(7);
+        }
+        if (url.href.includes('https://')) {
+          url_stub = url.href.substr(8);
+        }
+        switch (url.authentication.method) {
+          case 'jwt':
+            AUTH_DATA[url_stub] = ['jwt', url.authentication.token];
+            break;
+          case 'basic':
+            // AUTH_DATA[url_stub] = ['basic', ]
+            // eslint-disable-next-line no-fallthrough
+          case 'digest':
+            // AUTH_DATA[url_stub] = ['digest', ]
+            // eslint-disable-next-line no-fallthrough
+          default:
+            console.log(`${url.authentication.method} is not implemented`);
+            break;
+        }
+      }
+      adapter.loadThing(url.href);
```

Comment:
It seems to me that it would be easier to just pass the `url.authentication` object into `loadThing()`, rather than having this global `AUTH_DATA` object.

Reply:
I thought I needed this globally so I could get to it from all scopes.

It's where getHeaders gets the tokens to generate the auth string for rest and createWebsocket for the query param. 

As long as I can get to the tokens and base url this stuff could be stored anywhere. Kinda need to cache it if polling I think. Could go to the database for socket connections I guess.

Reply:
You could move the object and the function inside the adapter and then cache it by device ID.

Reply:
Cool -- ok -- so I added the getHeaders() function and authData property to ThingURLAdapter. In ThingURLDevice I store a reference to the adapter (didn't know how to get it from Device. 

I'm still loading authData in loadThingURLAdapter

Global function and AUTH_DATA removed. 

It all seems to work as we would like.